### PR TITLE
direct connection to acraserver with ssl

### PR DESCRIPTION
**Pull request by Lagovas**
* dropped hardcoded path to tls certificates
* added handling SIGINT to correct closing sockets (@mozhmike maybe will be useful in webui to handle SIGINT and call `server.Close()`)
* added test without acraproxy: app(tls) <-> acraserver(tls) <-> postgresql(tls)

because official postgresql image doesn't support ssl, I changed image to non-official with such support and added server.[key|crt] from image to `tests/` to use via acraserver.
and because postgresql support tls only over tcp (not over unix), and test with tls usage use TCP (with TCP TIME WAIT problem when port is binded even after killing process) I added changing ports between go versions

### CODE REVIEW

File: cmd/acraserver/acraserver.go [hunk 1]
```diff
@@ -16,14 +16,15 @@ package main
 import (
 	"crypto/tls"
 	"flag"
+	"net/http"
+	_ "net/http/pprof"
```
Comment (vixentael) [inline cmd/acraserver/acraserver.go]: is it used?

File: cmd/utils.go [hunk 2]
```diff
@@ -30,6 +33,39 @@ func init() {
...
+}
+
+func (handler *SignalHandler) AddCallback(callback SignalCallback) {
+	handler.callbacks = append(handler.callbacks, callback)
+}
+
+// Register should be called as goroutine
+func (handler *SignalHandler) Register() {
+	signal.Notify(handler.ch, handler.osSignal)
+	<-handler.ch
```
Comment (vixentael) [inline cmd/utils.go]: is it channel? https://tour.golang.org/concurrency/2

where it sends value from `handler.ch`?

File: tests/test.py [hunk 3]
```diff
@@ -166,28 +176,35 @@ def tearDownModule():
...
-
+    ACRA_PORT = int(os.environ.get('TEST_ACRA_PORT', 10003))
     ACRA_BYTEA = 'hex_bytea'
     DB_BYTEA = 'hex'
     WHOLECELL_MODE = False
     ZONE = False
     DEBUG_LOG = False
     TEST_DATA_LOG = False
     maxDiff = None
+    # hack to simplify handling errors on forks and don't check `if hasattr(self, 'proxy_1')`
```
Comment (vixentael) [inline tests/test.py]: 👍

File: cmd/utils.go (hunk 2 above)
Comment (Lagovas) [reply cmd/utils.go]: https://golang.org/pkg/os/signal/#Notify 
here we pass channel and block by reading until app receive signal

File: cmd/acraserver/acraserver.go (hunk 1 above)
Comment (Lagovas) [reply cmd/acraserver/acraserver.go]: yes, here - https://github.com/cossacklabs/acra/pull/112/files/591904efc2140f6b44cbc2d3ab8441e4c72811eb#diff-963cfdc2dd5ec26b3315f2be5214f061R169
for debugging golang provide way to fetch profiling data over http

**Comment by Lagovas** [comment]
close for remove noise of notifications

**Pull request closed without merging**